packages/web-components/src/theme/set-theme.ts

```diff
@@ -32,17 +35,23 @@ export const setTheme = (theme: Theme) => {
   if (!themeStyleTextMap.has(theme)) {
     const tokenDeclarations: string[] = [];
 
-    for (const t of tokenNames) {
+    for (const [tokenName, tokenValue] of Object.entries(theme)) {
+      if (typeof tokenValue !== 'string' && Number.isNaN(tokenValue)) {
+        throw new Error(
+          `The value of "${tokenName}" must be a string or a number, but got "${tokenValue.toString()}" instead.`,
+        );
+      }
+
       if (SUPPORTS_REGISTER_PROPERTY) {
         try {
           CSS.registerProperty({
-            name: `--${t}`,
+            name: `--${tokenName}`,
             inherits: true,
-            initialValue: theme[t] as string,
+            initialValue: tokenValue.toString(),
           });
         } catch {}
       }
-      tokenDeclarations.push(`--${t}: ${theme[t] as string};`);
+      tokenDeclarations.push(`--${tokenName}: ${tokenValue};`);
     }
 
     themeStyleTextMap.set(theme, `html{${tokenDeclarations.join('')}}`);
```

Comment:
Since this code is executed at run-time, I'm hesitant to add the checks, because the validation is a developer problem, it shouldn't be added to the perf load on the user end. I think we should have some validation mechanism in place at build-time at some point, especially given the upcoming semantic token support. I'm not sure where would this mechanism fit though.

Reply:
I improve the tsdoc and description in Storybook to make the requirement of theme object more clear. I also filed an issue for the potential build tool to validate theme objects: https://github.com/microsoft/fluentui/issues/32214.